SPI Loopback test fails to build on iMX RT EVKs

**Describe the bug**
The spi loopback test fails to build on the rt1010, rt1015, rt1020, and rt1050 evks

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -DBOARD=mimxrt1050_evk test/drivers/spi/spi_loopback
3. make
4. See error

**Expected behavior**
Build should succeed

**Logs and console output**
```
/home/danieldegrasse/zephyrproject/zephyr/drivers/dma/dma_mcux_edma.c:426:12: warning: 'dma_mcux_edma_init' defined but not used [-Wunused-function]
  426 | static int dma_mcux_edma_init(const struct device *dev)
      |            ^~~~~~~~~~~~~~~~~~
/home/danieldegrasse/zephyrproject/zephyr/drivers/dma/dma_mcux_edma.c:417:36: warning: 'dma_mcux_edma_api' defined but not used [-Wunused-const-variable=]
  417 | static const struct dma_driver_api dma_mcux_edma_api = {
      |                                    ^~~~~~~~~~~~~~~~~
/home/danieldegrasse/zephyrproject/zephyr/drivers/dma/dma_mcux_edma.c:162:13: warning: 'dma_mcux_edma_error_irq_handler' defined but not used [-Wunused-function]
  162 | static void dma_mcux_edma_error_irq_handler(const struct device *dev)
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/danieldegrasse/zephyrproject/zephyr/drivers/dma/dma_mcux_edma.c:143:13: warning: 'dma_mcux_edma_irq_handler' defined but not used [-Wunused-function]
  143 | static void dma_mcux_edma_irq_handler(const struct device *dev)
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~
[150/160] Linking C executable zephyr/zephyr_pre0.elf
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /home/danieldegrasse/zephyrproject/zephyr/build/zephyr/zephyr_pre0.map
: && ccache /opt/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc   zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/arm/core/offsets/offsets.c.obj  -fuse-ld=bfd  -Wl,-T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/home/danieldegrasse/zephyrproject/zephyr/build/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a  zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a  zephyr/arch/arch/arm/core/aarch32/mpu/libarch__arm__core__aarch32__mpu.a  zephyr/lib/libc/minimal/liblib__libc__minimal.a  zephyr/lib/posix/liblib__posix.a  zephyr/soc/arm/common/cortex_m/libsoc__arm__common__cortex_m.a  zephyr/boards/arm/mimxrt1010_evk/libboards__arm__mimxrt1010_evk.a  zephyr/subsys/testsuite/ztest/libsubsys__testsuite__ztest.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/dma/libdrivers__dma.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/spi/libdrivers__spi.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  modules/hal_nxp/hal_nxp/lib..__modules__hal__nxp.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L"/opt/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/thumb/v7e-m/nofp"  -L/home/danieldegrasse/zephyrproject/zephyr/build/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -no-pie  -mcpu=cortex-m7  -mthumb  -mabi=aapcs  -mfp16-format=ieee  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn && cd /home/danieldegrasse/zephyrproject/zephyr/build/zephyr && /usr/bin/cmake -E echo
/opt/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/spi/libdrivers__spi.a(spi_mcux_lpspi.c.obj):(.data.spi_mcux_data_0+0x98): undefined reference to `__device_dts_ord_33'
/opt/zephyr-sdk-0.13.1/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/10.3.0/../../../../arm-zephyr-eabi/bin/ld.bfd: zephyr/drivers/spi/libdrivers__spi.a(spi_mcux_lpspi.c.obj):(.data.spi_mcux_data_0+0xdc): undefined reference to `__device_dts_ord_33'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/danieldegrasse/zephyrproject/zephyr/build
```

--- a/boards/arm/mimxrt1010_evk/mimxrt1010_evk.dts
+++ b/boards/arm/mimxrt1010_evk/mimxrt1010_evk.dts
@@ -79,3 +79,7 @@ zephyr_udc0: &usb1 {
 &adc1 {
 	status = "okay";
 };
+
+&edma0 {
+	status = "okay";
+};

--- a/boards/arm/mimxrt1015_evk/mimxrt1015_evk.dts
+++ b/boards/arm/mimxrt1015_evk/mimxrt1015_evk.dts
@@ -108,3 +108,7 @@ zephyr_udc0: &usb1 {
 &adc1 {
 	status = "okay";
 };
+
+&edma0 {
+	status = "okay";
+};

--- a/boards/arm/mimxrt1020_evk/mimxrt1020_evk.dts
+++ b/boards/arm/mimxrt1020_evk/mimxrt1020_evk.dts
@@ -165,3 +165,7 @@ zephyr_udc0: &usb1 {
 &adc1 {
 	status = "okay";
 };
+
+&edma0 {
+	status = "okay";
+};

--- a/boards/arm/mimxrt1050_evk/mimxrt1050_evk.dts
+++ b/boards/arm/mimxrt1050_evk/mimxrt1050_evk.dts
@@ -227,3 +227,7 @@ zephyr_udc0: &usb1 {
 &wdog0 {
 	status = "okay";
 };
+
+&edma0 {
+	status = "okay";
+};

